net/grpc/helpers_test.go

```diff
@@ -0,0 +1,75 @@
+package grpc
+
+import (
+	"testing"
+)
+
+func Test_targetParser(t *testing.T) {
+	var tests = []struct {
+		input  string
+		result string
+	}{
+		{
+			input:  "",
+			result: ":",
+		},
+		{
+			input:  "127.0.0.1",
+			result: "127.0.0.1:" + defaultPort,
+		},
+		{
+			input:  "registry://127.0.0.1",
+			result: "127.0.0.1:" + defaultPort,
+		},
+		{
+			input:  "127.0.0.1:255",
+			result: "127.0.0.1:255",
+		},
+		{
+			input:  "registry://127.0.0.1:255",
+			result: "127.0.0.1:255",
+		},
+		{
+			input:  "hostname",
+			result: "hostname:" + defaultPort,
+		},
+		{
+			input:  "registry://hostname",
+			result: "hostname:" + defaultPort,
+		},
+		{
+			input:  "hostname:255",
+			result: "hostname:255",
+		},
+		{
+			input:  "registry://hostname:255",
+			result: "hostname:255",
+		},
```

Comment:
May be we should add `Hostname:255 -> hostname:255`?

Reply:
Added